### ISSUE TITLE
ios - keyboard showing throws textfield beyond AppBar

### ISSUE THREAD

**Issue by cosinus84** [flutter/flutter#20070]
## Steps to Reproduce

When SingleChildScrollView scrolled down, tap on textfield and the keyboard appears, the field   goes under the AppBar
MaterialApp->Scaffold>Stack>Container>Form>SingleChildScrollView(reverse true)>Column

```
quoteform_parent.dart
Scaffold(
              appBar: AppBar(
                  centerTitle: false,
                  title: const Text("MyApp"),
                  actions: <Widget>[
                    FlatButton(
                      textColor: Colors.white,
                      child: const Text("Logout"),
                      onPressed: () {
                        //@todo Clear isAuth state
                        Navigator.pushReplacementNamed(context, '/login');
                      },
                    )
                  ]),
              body: Stack(
                children: <Widget>[QuoteForm(), ConnectivityStatus()],
              ));
        });

quoteform.dart
Container(padding: EdgeInsets.only(top: 0.0, left: 16.0, right: 16.0), child:Form(
            key: _formKey,
            child: SingleChildScrollView(
              reverse: true,
              child:Column(children: <Widget>[....]
))
```

![1](https://user-images.githubusercontent.com/1459132/43511254-a0e4b524-9580-11e8-8e3f-c6e356f3b48d.png)
![1b](https://user-images.githubusercontent.com/1459132/43511255-a104145a-9580-11e8-98e5-abc0dee64f7a.png)
![1a](https://user-images.githubusercontent.com/1459132/43511256-a121a812-9580-11e8-840a-4db4e6e18d2c.png)

[✓] Flutter (Channel beta, v0.5.1, on Mac OS X 10.13.5 17F77, locale en-RO)
    • Flutter version 0.5.1 at /Users/cosinus/flutter
    • Framework revision c7ea3ca377 (9 weeks ago), 2018-05-29 21:07:33 +0200
    • Engine revision 1ed25ca7b7
    • Dart version 2.0.0-dev.58.0.flutter-f981f09760

**Comment by cosinus84** [comment]
Actually I think this is other kind of issue. Is correct behaviour of scaffold (resizeToAvoidBottomPadding: true) . For Add client name and Add Advisor name works because even if the keyboard pushes the Container/Form(View) up, the fields are at the bottom so they are still visible. Because the amount field is near the top when the keyboard appears it pushes the Container/Form(View) so the field goes up under AppBar. 

*Still don't understand why SingleChildScrollView with reverse: true works. With reverse:false keyboard doesn't push all up.

What are the strategies with this kind of forms, I somehow think anyone doing forms meet this problem.

**Comment by zoechi** [comment]
Did you close accidentally?
If you expect an answer it's better to keep it open.

**Comment by cosinus84** [comment]
Yes I closed it because I found a solution that scrolls the SingleChildScrollView: https://www.didierboelens.com/2018/04/hint-4-ensure-a-textfield-or-textformfield-is-visible-in-the-viewport-when-has-the-focus/ 

As I said in a comment earlier, seems not too be an issue, is the correct behaviour when using my original code(it just pushed the view, up by keyboard height).

The actual "issue" is that flutter doesn't provide a solution for this kind of problem: a form with inputs in a scrollview, tap on an input, keyboard hides the input. I think most of apps have a form in a scrollview (ex. iphone 5s)

RN has the same unfinished problem, I hopped that we escaped this nightmare :) with flutter
